### PR TITLE
freq_to_period for 'D'

### LINKED ISSUE #2109: freq_to_period doens't understand "D"
I'm trying to compute some nice seasonal_decompose of a 'D' freq'd series, but this freq isn't recognized by https://github.com/statsmodels/statsmodels/blob/master/statsmodels/tsa/tsatools.py#L625... Is there a good reason for it ?

**edit**  doesn't understand 'H' neither (see comment and link to stackoverflow below)

Best !

Thomas

### PR DESCRIPTION
**Pull request by dhpiyush**
fix #2109

### CODE REVIEW

**Comment by josef-pkt** [comment]
@dhpiyush   Thanks, I think it looks good, but I'm not really familiar with this part. I will go over the issue/PR backlog soon.

Can you add a unit test for the frequency?   There might already be cases in the test suite that could be copied and/or adjusted, or http://stackoverflow.com/questions/34494780/time-series-analysis-unevenly-spaced-measures-pandas-statsmodels might provide an example.

**Comment by dhpiyush** [comment]
>>> x = np.array([-50, 175, 149, 214, 247, 237, 225, 329, 729, 809,530, 489, 540, 457, 195, 176, 337, 239, 128, 102,232, 429, 3, 98, 43, -141, -77, -13, 125, 361, -45, 184,1,1,1,1,23,23,2,32,32,32,32,32,32,3,23,23,2,3,23,2,32,3,23,23,2,32,323,2,32,32,32,3,23,23,23,2,32,3,23,2,32,3,2323,2,323,2,32,32,3,23,23,23,23,2,32,32,3,23,2,3,23,23,3,4,34,34,3,43,4,34,34,3,4,34,3,4,34,3,43,4,34,34,3,43,56,56,56,5,65,65,65,6,56,56,5,65,65,6,56,5,65,6,56,5,6,56,56,56,56,56,5,65,65,6,56,56,5,65,6,56,56,5,6,56,56,65,6,76,7,67,6,76,7,67,6,7,67,67,6,7,67,6,7,67,6,7,67,67,87,54,5,45,4,54,5,45,45,45,64,64,64,64,64,6,46,46,46,4,64,64,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,5,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,5,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,5,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,5,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,9])
>>> data = DataFrame(x, DatetimeIndex(start='1/1/1951',
...                                       periods=len(x),
...                                       freq='D'))
>>> res = seasonal_decompose(data)
>>> res.resid
             0
1951-01-01 NaN
1951-01-02 NaN
1951-01-03 NaN
1951-01-04 NaN
1951-01-05 NaN
1951-01-06 NaN
1951-01-07 NaN
1951-01-08 NaN
1951-01-09 NaN
1951-01-10 NaN
1951-01-11 NaN
1951-01-12 NaN
1951-01-13 NaN
1951-01-14 NaN
1951-01-15 NaN
1951-01-16 NaN
1951-01-17 NaN
1951-01-18 NaN
1951-01-19 NaN
1951-01-20 NaN
1951-01-21 NaN
1951-01-22 NaN
1951-01-23 NaN
1951-01-24 NaN
1951-01-25 NaN
1951-01-26 NaN
1951-01-27 NaN
1951-01-28 NaN
1951-01-29 NaN
1951-01-30 NaN
...         ..
1953-01-18 NaN
1953-01-19 NaN
1953-01-20 NaN
1953-01-21 NaN
1953-01-22 NaN
1953-01-23 NaN
1953-01-24 NaN
1953-01-25 NaN
1953-01-26 NaN
1953-01-27 NaN
1953-01-28 NaN
1953-01-29 NaN
1953-01-30 NaN
1953-01-31 NaN
1953-02-01 NaN
1953-02-02 NaN
1953-02-03 NaN
1953-02-04 NaN
1953-02-05 NaN
1953-02-06 NaN
1953-02-07 NaN
1953-02-08 NaN
1953-02-09 NaN
1953-02-10 NaN
1953-02-11 NaN
1953-02-12 NaN
1953-02-13 NaN
1953-02-14 NaN
1953-02-15 NaN
1953-02-16 NaN

[778 rows x 1 columns]

This works only when number of elements in  array x is greater than or equal to 365

File: statsmodels/graphics/tsaplots.py [hunk 1]
```diff
@@ -177,7 +177,12 @@ def plot_pacf(x, ax=None, lags=None, alpha=.05, method='ywm',
     kwargs.setdefault('linestyle', 'None')
     ax.margins(.05)
     ax.plot(lags, acf_x, **kwargs)
-    ax.set_title("Partial Autocorrelation")
+
+    if title is None:
+
+        ax.set_title("Partial Autocorrelation")
+    else:
+        ax.set_title(title)
```
Comment (jbrockmendel) [inline statsmodels/graphics/tsaplots.py]: Is this related to the rest of the PR?

**Comment by bashtage** [comment]
frequency 'D' is 7 in master and so this PR cannot go forward.
